librz/main/rz-ax.c

```diff
@@ -412,6 +415,35 @@ static int rax(RzNum *num, char *str, int len, int last, ut64 *_flags, int *fm)
 				ch & 1 ? 1 : 0);
 		}
 		return true;
+	} else if (has_flag(flags, RZ_AX_FLAG_SET_BITS)) { // -p (count set bits)
+		ut64 n = rz_num_math(num, str);
+		char strbits[65];
+		int i = 0, set_bits_ctr = 0;
+		rz_num_to_bits(strbits, n);
+		rz_str_reverse(strbits); // because we count Right to Left
+		while (strbits[i] != '\0') {
+			if (strbits[i] == '1') {
+				++set_bits_ctr;
+				if (i == 0) {
+					printf("[%d", i);
+				} else if (strbits[i] == '1' && strbits[i - 1] == '0') {
+					printf("[%d", i);
+				}
+			}
+			if (strbits[i] == '0' && strbits[i - 1] == '1') {
+				if (set_bits_ctr == 1) {
+					printf("]: 1\n");
+				} else
+					printf("-%d]: 1\n", i - 1);
+				set_bits_ctr = 0;
+			} else if (strbits[i] == '1' && strbits[i + 1] == '\0') {
+				printf("]: 1\n");
+				set_bits_ctr = 0;
+			}
+
+			++i;
+		}
+		return true;
```

Comment:
if it counts, why it does not just output the total number? also you do not really need to convert the number to bits, since you can just loop and shift by 1 till the value is zero.
```c
ut64 n = rz_num_math(num, str);
ut32 n_bits = 0;
while (n) {
    n_bits += n & 1;
    n >>= 1;
}
printf("%u\n", n);
```

Reply:
ok, i did not see the issue and the output. i think the description of this mode is wrong, since it shows where and how many bits are in a value. @XVilka is better to change the name to something like `counts the number of bits in groups`, because if you describe `-p` as `count number of bits` i expect to just output the total number of ones like in the algo above.

Reply:
Changing it to ``  -p      position of set bits ;  rz-ax -p 0xb3\n");
`` since that is what it actually does as has been described in linked issue

Reply:
sounds even better